CLIENT_SPECIFICATION.md

```diff
@@ -0,0 +1,194 @@
+
+1. `windows` - Not available
+2. `common` - Not available
+3. `osx` - Not available
+4. `linux` - Page found
+
+Steps #3 and #4 may be done in either order.
+
+#### If a page isn't found
+If a page cannot be found in _any_ platform, then it is RECOMMENDED that clients display an error message with a link to create a new issue against the `tldr-pages/tldr` GitHub repository. Said link might take the following form:
+
+```url
+https://github.com/tldr-pages/tldr/issues/new?title=page%20request:%20{command_name}
+```
+
+....where `{command_name}` is the name of the command that was not found.
+
+#### If multiple versions of a page were found
+If multiple versions of a page were found for different platforms, then a client MAY choose to display a notice to the user notifying them of this.
+
+## Language
+Pages can be written in multiple languages. If a client has access to environment variables, several standard ones exist to specify the language in which a client should operate. If not, then clients MUST make reasonable assumptions based on the information provided by the environment in which they operate (e.g. consulting `navigator.languages` in a browser, etc.).
+
+The [`LANG` environment variable](https://www.gnu.org/software/gettext/manual/html_node/Locale-Environment-Variables.html) MUST be used to determine both the language of a page to display, and the display language of any interface text shown to the user (e.g. things like `Cache updated successfully`, for example).
+
+Additionally, the [`LC_MESSAGES` environment variable](https://www.gnu.org/software/gettext/manual/html_node/Locale-Environment-Variables.html) MAY be present. If specified, clients MUST use it's value to determine the language in which interface text is shown in (separately from the language to display the page in).
+
+Finally, the [`LANGUAGE` environment variable](https://www.gnu.org/software/gettext/manual/html_node/The-LANGUAGE-variable.html) specifies a priority list of languages that a user wishes to read in. If present, a client MUST use the defined priority list to decide on the language that to display in.
+
+If a page is not available in the user's preferred language, then a client MUST respect the user's priority list defined in the `LANGUAGE` variable (if specified), and MAY choose to notify the user that a page in their chosen language couldn't be found (perhaps along with a link to the [translations section of the contributing guide](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md#translations)).
+
+## Other Considerations
+This section contains a number of other items that don't neatly fit into any of the sections defined above.
+
+If appropriate, it is RECOMMENDED that clients implement a cache of pages. If implemented, clients MUST download the _entire_ archive from **UNKNOWN_LOCATION**.
+
+Additionally, clients MAY automatically the cache on a regular basis.
```

Comment:
May automatically _update_ the cache

Reply:
Some hopefully helpful comments:
* Why is it called `tldr-pages`? Is the name `tldr` reserved for something else??
* `It is _not_ a specification of the format of the pages themselves` -- I am guessing you mean the output format, but this could also be taken to refer to the storage format for the pages, so a clarifying word would be helpful.
* Why MUST `-a`/`--list-all` list only one page per line? I think this is usually not what people want for ease of use. I would prefer an additional separate commandline option `-n`/`--no-formatting` that lists everything plain: no colours, only 1 line per page in a listing.
* If this is **only a specification of how a user should be able to interface with an official client** then why specify the `Directory Structure`? That should be irrelevant, and in no need of specification. Same for `Translation`. But if it is important to include these sections, better adjust the language in the introduction, so that it is not only about the interface.